In [1]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 

In [2]:
df = pd.read_csv("C:/Users/burnw/OneDrive/Documents/coding/Data Analyst/Journey/Week 6-7/data/post_pandemic_remote_work_health_impact_2025.csv")
df.head()

Survey_Date  Age  Gender         Region               Industry  \
0  2025-06-01   27  Female           Asia  Professional Services   
1  2025-06-01   37  Female           Asia  Professional Services   
2  2025-06-01   32  Female         Africa              Education   
3  2025-06-01   40  Female         Europe              Education   
4  2025-06-01   30    Male  South America          Manufacturing   

           Job_Role Work_Arrangement  Hours_Per_Week Mental_Health_Status  \
0      Data Analyst           Onsite              64      Stress Disorder   
1      Data Analyst           Onsite              37      Stress Disorder   
2  Business Analyst           Onsite              36                 ADHD   
3      Data Analyst           Onsite              63                 ADHD   
4   DevOps Engineer           Hybrid              65                  NaN   

  Burnout_Level  Work_Life_Balance_Score     Physical_Health_Issues  \
0          High                        3   Shoulder Pain; Neck Pain   
1          High                        4                  Back Pain   
2          High                        3  Shoulder Pain; Eye Strain   
3        Medium                        1  Shoulder Pain; Eye Strain   
4        Medium                        5                        NaN   

   Social_Isolation_Score Salary_Range  
0                       2     $40K-60K  
1                       2    $80K-100K  
2                       2    $80K-100K  
3                       2     $60K-80K  
4                       4     $60K-80K

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3157 entries, 0 to 3156
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Survey_Date              3157 non-null   object
 1   Age                      3157 non-null   int64 
 2   Gender                   3157 non-null   object
 3   Region                   3157 non-null   object
 4   Industry                 3157 non-null   object
 5   Job_Role                 3157 non-null   object
 6   Work_Arrangement         3157 non-null   object
 7   Hours_Per_Week           3157 non-null   int64 
 8   Mental_Health_Status     2358 non-null   object
 9   Burnout_Level            3157 non-null   object
 10  Work_Life_Balance_Score  3157 non-null   int64 
 11  Physical_Health_Issues   2877 non-null   object
 12  Social_Isolation_Score   3157 non-null   int64 
 13  Salary_Range             3157 non-null   object
dtypes: int64(4), object(10)
memory usage: 34

In [4]:
df.isnull().sum()

Survey_Date                  0
Age                          0
Gender                       0
Region                       0
Industry                     0
Job_Role                     0
Work_Arrangement             0
Hours_Per_Week               0
Mental_Health_Status       799
Burnout_Level                0
Work_Life_Balance_Score      0
Physical_Health_Issues     280
Social_Isolation_Score       0
Salary_Range                 0
dtype: int64

In [5]:
df['Mental_Health_Status'].unique()

array(['Stress Disorder', 'ADHD', nan, 'Burnout', 'Anxiety', 'PTSD',
       'Depression'], dtype=object)

In [6]:
df['Industry'].value_counts()

Industry
Professional Services    730
Technology               593
Manufacturing            370
Finance                  367
Education                287
Healthcare               247
Marketing                202
Retail                   188
Customer Service         173
Name: count, dtype: int64

In [7]:
len(df['Job_Role'].unique())

24

In [8]:
# Drop the survey date and job role column
df.drop(columns=['Survey_Date', 'Job_Role'], axis=1, inplace=True)
df.head()

Age  Gender         Region               Industry Work_Arrangement  \
0   27  Female           Asia  Professional Services           Onsite   
1   37  Female           Asia  Professional Services           Onsite   
2   32  Female         Africa              Education           Onsite   
3   40  Female         Europe              Education           Onsite   
4   30    Male  South America          Manufacturing           Hybrid   

   Hours_Per_Week Mental_Health_Status Burnout_Level  Work_Life_Balance_Score  \
0              64      Stress Disorder          High                        3   
1              37      Stress Disorder          High                        4   
2              36                 ADHD          High                        3   
3              63                 ADHD        Medium                        1   
4              65                  NaN        Medium                        5   

      Physical_Health_Issues  Social_Isolation_Score Salary_Range  
0   Shoulder Pain; Neck Pain                       2     $40K-60K  
1                  Back Pain                       2    $80K-100K  
2  Shoulder Pain; Eye Strain                       2    $80K-100K  
3  Shoulder Pain; Eye Strain                       2     $60K-80K  
4                        NaN                       4     $60K-80K

In [9]:
df['Salary_Range'].unique()

array(['$40K-60K', '$80K-100K', '$60K-80K', '$100K-120K', '$120K+'],
      dtype=object)

In [10]:
a = '$40K-60K'

salary_range = a.replace('K', '000').replace('$', '').split('-')
print(salary_range)
salary_range = [int(s) for s in salary_range]
print(salary_range)
average_salary = sum(salary_range) / len(salary_range)
print(average_salary)

isinstance(average_salary, float) # --> True
isinstance(average_salary, int) # --> False


['40000', '60000']
[40000, 60000]
50000.0


False

In [11]:
# Convert Salary range to average salary 

df['Salary_Range'] = df['Salary_Range'].str.replace('K', '000').str.replace('$', '').str.replace('+', '').str.split('-')
df['Salary_Range']

0         [40000, 60000]
1        [80000, 100000]
2        [80000, 100000]
3         [60000, 80000]
4         [60000, 80000]
              ...       
3152     [80000, 100000]
3153    [100000, 120000]
3154      [40000, 60000]
3155     [80000, 100000]
3156    [100000, 120000]
Name: Salary_Range, Length: 3157, dtype: object

In [12]:
df['Salary_Range'].value_counts()

Salary_Range
[60000, 80000]      1014
[80000, 100000]      964
[40000, 60000]       510
[100000, 120000]     459
[120000]             210
Name: count, dtype: int64

In [13]:
df['Salary_Range'] = df['Salary_Range'].apply(lambda x: [int(i) for i in x] if isinstance(x, list) else x)
df['Avg_Salary'] = df['Salary_Range'].apply(lambda x: sum(x) / len(x) if isinstance(x, list) else x)
df['Avg_Salary']

0        50000.0
1        90000.0
2        90000.0
3        70000.0
4        70000.0
          ...   
3152     90000.0
3153    110000.0
3154     50000.0
3155     90000.0
3156    110000.0
Name: Avg_Salary, Length: 3157, dtype: float64

In [14]:
# Drop the Salary_Range column 
df.drop(columns='Salary_Range', axis=1, inplace=True)

In [15]:
df.head()

Age  Gender         Region               Industry Work_Arrangement  \
0   27  Female           Asia  Professional Services           Onsite   
1   37  Female           Asia  Professional Services           Onsite   
2   32  Female         Africa              Education           Onsite   
3   40  Female         Europe              Education           Onsite   
4   30    Male  South America          Manufacturing           Hybrid   

   Hours_Per_Week Mental_Health_Status Burnout_Level  Work_Life_Balance_Score  \
0              64      Stress Disorder          High                        3   
1              37      Stress Disorder          High                        4   
2              36                 ADHD          High                        3   
3              63                 ADHD        Medium                        1   
4              65                  NaN        Medium                        5   

      Physical_Health_Issues  Social_Isolation_Score  Avg_Salary  
0   Shoulder Pain; Neck Pain                       2     50000.0  
1                  Back Pain                       2     90000.0  
2  Shoulder Pain; Eye Strain                       2     90000.0  
3  Shoulder Pain; Eye Strain                       2     70000.0  
4                        NaN                       4     70000.0

---

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3157 entries, 0 to 3156
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Age                      3157 non-null   int64  
 1   Gender                   3157 non-null   object 
 2   Region                   3157 non-null   object 
 3   Industry                 3157 non-null   object 
 4   Work_Arrangement         3157 non-null   object 
 5   Hours_Per_Week           3157 non-null   int64  
 6   Mental_Health_Status     2358 non-null   object 
 7   Burnout_Level            3157 non-null   object 
 8   Work_Life_Balance_Score  3157 non-null   int64  
 9   Physical_Health_Issues   2877 non-null   object 
 10  Social_Isolation_Score   3157 non-null   int64  
 11  Avg_Salary               3157 non-null   float64
dtypes: float64(1), int64(4), object(7)
memory usage: 296.1+ KB


In [17]:
df.Gender.value_counts()

Gender
Male                 1535
Female               1500
Non-binary             90
Prefer not to say      32
Name: count, dtype: int64

In [18]:
df['Gender'] = df['Gender'].str.strip().str.title()
df[df['Gender'] == 'Non-Binary']

# Sorry, my mistake, I mistyped the word which i want to filter, this works perfectly:
# df[df['Gender'] == 'Non-binary'] 

Age      Gender         Region               Industry Work_Arrangement  \
7      63  Non-Binary         Europe  Professional Services           Onsite   
138    37  Non-Binary  South America             Technology           Onsite   
149    46  Non-Binary         Europe  Professional Services           Hybrid   
175    60  Non-Binary  South America       Customer Service           Onsite   
191    47  Non-Binary         Africa             Technology           Hybrid   
...   ...         ...            ...                    ...              ...   
2955   23  Non-Binary  South America             Technology           Remote   
2960   38  Non-Binary        Oceania             Healthcare           Remote   
2984   27  Non-Binary         Africa              Marketing           Hybrid   
2994   56  Non-Binary           Asia              Marketing           Onsite   
3043   63  Non-Binary        Oceania                Finance           Onsite   

      Hours_Per_Week Mental_Health_Status Burnout_Level  \
7                 55              Anxiety          High   
138               48              Anxiety           Low   
149               45                  NaN           Low   
175               51           Depression          High   
191               64              Anxiety           Low   
...              ...                  ...           ...   
2955              42           Depression        Medium   
2960              39                  NaN          High   
2984              63                 ADHD           Low   
2994              63                 PTSD        Medium   
3043              51              Burnout          High   

      Work_Life_Balance_Score                Physical_Health_Issues  \
7                           3                                   NaN   
138                         1                                   NaN   
149                         5                         Shoulder Pain   
175                         3   Back Pain; Shoulder Pain; Neck Pain   
191                         3                                   NaN   
...                       ...                                   ...   
2955                        5                         Shoulder Pain   
2960                        1                            Eye Strain   
2984                        3                            Wrist Pain   
2994                        4                 Eye Strain; Neck Pain   
3043                        4  Back Pain; Shoulder Pain; Eye Strain   

      Social_Isolation_Score  Avg_Salary  
7                          2    110000.0  
138                        3     90000.0  
149                        1     90000.0  
175                        2     70000.0  
191                        1     70000.0  
...                      ...         ...  
2955                       5     50000.0  
2960                       3    120000.0  
2984                       3     90000.0  
2994                       1     90000.0  
3043                       2     50000.0  

[90 rows x 12 columns]

In [19]:
df[df['Gender'].isin(['Prefer Not To Say', 'Non-Binary'])]

Age             Gender         Region               Industry  \
7      63         Non-Binary         Europe  Professional Services   
138    37         Non-Binary  South America             Technology   
149    46         Non-Binary         Europe  Professional Services   
175    60         Non-Binary  South America       Customer Service   
191    47         Non-Binary         Africa             Technology   
...   ...                ...            ...                    ...   
2984   27         Non-Binary         Africa              Marketing   
2994   56         Non-Binary           Asia              Marketing   
3004   35  Prefer Not To Say  South America  Professional Services   
3043   63         Non-Binary        Oceania                Finance   
3107   43  Prefer Not To Say         Africa              Education   

     Work_Arrangement  Hours_Per_Week Mental_Health_Status Burnout_Level  \
7              Onsite              55              Anxiety          High   
138            Onsite              48              Anxiety           Low   
149            Hybrid              45                  NaN           Low   
175            Onsite              51           Depression          High   
191            Hybrid              64              Anxiety           Low   
...               ...             ...                  ...           ...   
2984           Hybrid              63                 ADHD           Low   
2994           Onsite              63                 PTSD        Medium   
3004           Onsite              44              Burnout        Medium   
3043           Onsite              51              Burnout          High   
3107           Onsite              60                  NaN           Low   

      Work_Life_Balance_Score  \
7                           3   
138                         1   
149                         5   
175                         3   
191                         3   
...                       ...   
2984                        3   
2994                        4   
3004                        4   
3043                        4   
3107                        2   

                               Physical_Health_Issues  Social_Isolation_Score  \
7                                                 NaN                       2   
138                                               NaN                       3   
149                                     Shoulder Pain                       1   
175               Back Pain; Shoulder Pain; Neck Pain                       2   
191                                               NaN                       1   
...                                               ...                     ...   
2984                                       Wrist Pain                       3   
2994                            Eye Strain; Neck Pain                       1   
3004  Back Pain; Shoulder Pain; Neck Pain; Wrist Pain                       4   
3043             Back Pain; Shoulder Pain; Eye Strain                       2   
3107                            Back Pain; Eye Strain                       3   

      Avg_Salary  
7       110000.0  
138      90000.0  
149      90000.0  
175      70000.0  
191      70000.0  
...          ...  
2984     90000.0  
2994     90000.0  
3004     90000.0  
3043     50000.0  
3107    110000.0  

[122 rows x 12 columns]

As my goal is **training an ML model**, and class imbalance is extreme:

- The data have only **122 out of 3157** (≈3.8%) in these two categories.
- This might not generalize well in model training (especially for categorical encoders).

✅ Then dropping them is acceptable:

> “The categories ‘Non-Binary’ and ‘Prefer not to say’ (≈3.8% of total data) were excluded from training to prevent model bias from extreme class imbalance.”


In [20]:
df.drop(df[df['Gender'].isin(['Non-Binary', 'Prefer Not To Say'])].index, inplace=True)

In [21]:
df['Gender'].value_counts()

Gender
Male      1535
Female    1500
Name: count, dtype: int64

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3035 entries, 0 to 3156
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Age                      3035 non-null   int64  
 1   Gender                   3035 non-null   object 
 2   Region                   3035 non-null   object 
 3   Industry                 3035 non-null   object 
 4   Work_Arrangement         3035 non-null   object 
 5   Hours_Per_Week           3035 non-null   int64  
 6   Mental_Health_Status     2273 non-null   object 
 7   Burnout_Level            3035 non-null   object 
 8   Work_Life_Balance_Score  3035 non-null   int64  
 9   Physical_Health_Issues   2768 non-null   object 
 10  Social_Isolation_Score   3035 non-null   int64  
 11  Avg_Salary               3035 non-null   float64
dtypes: float64(1), int64(4), object(7)
memory usage: 308.2+ KB


In [23]:
print(df['Mental_Health_Status'].isnull().sum(),
df['Physical_Health_Issues'].isnull().sum())

762 267


In [24]:
df['Mental_Health_Status'].value_counts() 


Mental_Health_Status
PTSD               409
ADHD               376
Anxiety            374
Burnout            373
Depression         372
Stress Disorder    369
Name: count, dtype: int64

In [25]:
df['Physical_Health_Issues'].value_counts() 

Physical_Health_Issues
Shoulder Pain; Eye Strain                                      256
Back Pain                                                      247
Back Pain; Eye Strain                                          246
Back Pain; Shoulder Pain; Eye Strain                           245
Eye Strain                                                     243
Shoulder Pain                                                  223
Back Pain; Shoulder Pain                                       216
Neck Pain                                                       91
Back Pain; Eye Strain; Neck Pain                                89
Back Pain; Shoulder Pain; Eye Strain; Neck Pain                 88
Back Pain; Neck Pain                                            85
Eye Strain; Neck Pain                                           79
Shoulder Pain; Neck Pain                                        74
Shoulder Pain; Eye Strain; Neck Pain                            71
Back Pain; Shoulder Pain; Neck Pain    

In [26]:
# Physical Health issues
df['Physical_Health_Issues'] = df['Physical_Health_Issues'].fillna('')

In [27]:
a = 'Back Pain; Shoulder Pain; Eye Strain'
len(a.split(';'))

3

In [28]:
df['Physical_Health_Count'] = df['Physical_Health_Issues'].apply(
    lambda x: len([issue for issue in x.split(';') if issue.strip() != ''])
    )

In [29]:
df.drop(columns='Physical_Health_Issues', axis=1, inplace=True)

In [30]:
df['Mental_Health_Status'].isnull().sum()

762

In [31]:
df['Mental_Health_Status'].unique()

array(['Stress Disorder', 'ADHD', nan, 'Burnout', 'Anxiety', 'PTSD',
       'Depression'], dtype=object)

In [32]:
# Mental Health issue

df['Mental_Health_Status'].mode()

# Why Filling All with Mode (“PTSD”) Isn't Ideal

# If you impute all missing values with "PTSD", you'll artificially inflate that category —
# suddenly PTSD will look like the most common condition, even though in reality it might not be.

# 👉 This creates data bias and distorts relationships with your target (Burnout_Level).
# It's basically saying “everyone who didn't answer has PTSD,” which is obviously not true.

0    PTSD
Name: Mental_Health_Status, dtype: object

In [33]:
df['Mental_Health_Status'] = df['Mental_Health_Status'].fillna('Unknown')

The mental health labels **do** represent increasing levels of difficulty or severity (loosely, but reasonably):

| Status | Interpretation |
| --- | --- |
| **Stress Disorder** | Moderate, short-term condition |
| **ADHD** | Chronic but manageable |
| **Burnout** | Prolonged fatigue and emotional exhaustion |
| **Anxiety** | Persistent worry; psychological symptoms |
| **Depression** | Severe emotional and mental fatigue |
| **PTSD** | Long-term traumatic stress |
| **Unknown** | No data reported |

So you can think of it as **a gradient of psychological impact**, from “mild stress” to “severe trauma.”

In [34]:
mh_map = {
    'Unknown': 0,
    'ADHD': 1,
    'Stress Disorder': 2,
    'Anxiety': 3,
    'Burnout': 4,
    'Depression': 5,
    'PTSD': 6
}

df['Mental_Health_Score'] = df['Mental_Health_Status'].map(mh_map)

In [35]:
df.drop(columns='Mental_Health_Status', axis=1, inplace=True)

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3035 entries, 0 to 3156
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Age                      3035 non-null   int64  
 1   Gender                   3035 non-null   object 
 2   Region                   3035 non-null   object 
 3   Industry                 3035 non-null   object 
 4   Work_Arrangement         3035 non-null   object 
 5   Hours_Per_Week           3035 non-null   int64  
 6   Burnout_Level            3035 non-null   object 
 7   Work_Life_Balance_Score  3035 non-null   int64  
 8   Social_Isolation_Score   3035 non-null   int64  
 9   Avg_Salary               3035 non-null   float64
 10  Physical_Health_Count    3035 non-null   int64  
 11  Mental_Health_Score      3035 non-null   int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 308.2+ KB


In [37]:
df['Burnout_Level'].value_counts()

Burnout_Level
Medium    1319
High      1005
Low        711
Name: count, dtype: int64

In [38]:
bl_map = {
    'High' : 3,
    'Medium' : 2,
    'Low' : 1
}

df['Burnout_Level'] = df['Burnout_Level'].map(bl_map)

In [39]:
cat_features = df.select_dtypes(include=object).columns
cat_features

Index(['Gender', 'Region', 'Industry', 'Work_Arrangement'], dtype='object')

In [40]:
num_features = df.select_dtypes(exclude=object).columns
num_features

Index(['Age', 'Hours_Per_Week', 'Burnout_Level', 'Work_Life_Balance_Score',
       'Social_Isolation_Score', 'Avg_Salary', 'Physical_Health_Count',
       'Mental_Health_Score'],
      dtype='object')

In [41]:
df.to_csv('clean_data.csv', index=False)